src/plugins/analytics_plugin.py

```diff
@@ -1,179 +1,150 @@
-
-    RT = TypeVar("RT")
-
-    def find_session_idx(func: Callable[..., RT]) -> int:
-        """Find session index in function call parameter."""
-        func_params = signature(func).parameters
-        try:
-            # func_params is an ordered dict -- this is the "recommended" way of getting the position
-            session_args_idx = tuple(func_params).index("session")
-        except ValueError:
-            raise ValueError(f"Function {func.__qualname__} has no `session` argument") from None
-
-        return session_args_idx
-
-    @contextlib.contextmanager
-    def create_session():
-        """Contextmanager that will create and teardown a session."""
-        session = settings.Session
-        try:
-            yield session
-            session.commit()
-        except Exception:
-            session.rollback()
-            raise
-        finally:
-            session.close()
-
-    def provide_session(func: Callable[..., RT]) -> Callable[..., RT]:
-        """
-        Function decorator that provides a session if it isn't provided.
-        If you want to reuse a session or run the function as part of a
-        database transaction, you pass it to the function, if not this wrapper
-        will create one and close it for you.
-        """
-        session_args_idx = find_session_idx(func)
-
-        @wraps(func)
-        def wrapper(*args, **kwargs) -> RT:
-            if "session" in kwargs or session_args_idx < len(args):
-                return func(*args, **kwargs)
-            else:
-                with create_session() as session:
-                    return func(*args, session=session, **kwargs)
-
-        return wrapper
+airflow_webserver_base_url = configuration.get("webserver", "BASE_URL")
+
 
 @provide_session
-def dags_report(session) -> Any:
-    start_date = request.args.get("startDate") # 2022-08-01
-    end_date = request.args.get("endDate") #2022-08-30
+def tasks_report(session) -> Any:
+    untrusted_start_date = request.args.get("startDate")  # 2022-08-01
+    untrusted_end_date = request.args.get("endDate")  # 2022-08-30
     try:
-        parse(start_date)
-        parse(end_date)
+        start_date = parse(untrusted_start_date)
+        end_date = parse(untrusted_end_date)
+    except ValueError as e:
+        log.error(f"The string is not a date: {e}")
+        raise e
+    else:
+        log_success = aliased(Log, name="log_success")
+        log_failed = aliased(Log, name="log_failed")
         # remove the dummy operator and the astronomer_monitoring_dag that is added in runtime from task count
-        sql = text(
-            f"""
-            select event, count(*) as totalCount from log l join task_instance ti on event in ('success', 'failed') and l.dttm >= '{start_date}' and l.dttm <= '{end_date}' and ti.operator != 'DummyOperator' and l.dag_id != 'astronomer_monitoring_dag' and ti.task_id = l.task_id group by event
-        """
+        query = (
+            session.query(
```

Comment:
can we refactor query back to one query with one join, we had optimized it to one query instead of this two query approach with 2 joins, is there a reason why we went this way?

Reply:
Reverted. :)